gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/GaaSObservabilityEventProducer.java

```diff
@@ -105,19 +107,19 @@ private GaaSObservabilityEventExperimental createGaaSObservabilityEvent(final St
     builder.setTimestamp(System.currentTimeMillis())
         .setFlowName(jobState.getProp(TimingEvent.FlowEventConstants.FLOW_NAME_FIELD))
         .setFlowGroup(jobState.getProp(TimingEvent.FlowEventConstants.FLOW_GROUP_FIELD))
+        .setFlowGraphEdgeId(jobState.getProp(TimingEvent.FlowEventConstants.FLOW_EDGE_FIELD))
         .setFlowExecutionId(jobState.getPropAsLong(TimingEvent.FlowEventConstants.FLOW_EXECUTION_ID_FIELD))
         .setJobName(jobState.getProp(TimingEvent.FlowEventConstants.JOB_NAME_FIELD))
         .setExecutorUrl(jobState.getProp(TimingEvent.METADATA_MESSAGE))
+        .setExecutorId(jobState.getProp(TimingEvent.FlowEventConstants.SPEC_EXECUTOR_FIELD))
         .setJobStartTime(jobStartTime)
         .setJobEndTime(jobEndTime)
+        .setJobOrchestratedTime(jobOrchestratedTime)
         .setIssues(issueList)
         .setJobStatus(status)
-        // TODO: Populate the below fields in a separate PR
-        .setExecutionUserUrn(null)
-        .setExecutorId("")
-        .setLastFlowModificationTime(0)
-        .setFlowGraphEdgeId("")
-        .setJobOrchestratedTime(null); // TODO: Investigate why TimingEvent.JOB_ORCHESTRATED_TIME is never propagated to the JobStatus
+        .setExecutionUserUrn(userToProxy)
+        // TODO: Set this value when flowSpec stores its modification time
+        .setLastFlowModificationTime(0);
```

Comment:
can you extract this value from the spec config? that's where I am storing it

Reply:
I sort of can in a very roundabout way, I'd send it as part of job properties and bubble it up. I'll wait for your PR to go in first though